src/pathfinding/grid_pathfinder.jl

```diff
@@ -0,0 +1,260 @@
+export CostMetric, MooreMetric, VonNeumannMetric, EuclideanMetric, ChebyshevMetric, HeightMapMetric, Pathfinder, Path, delta_cost, find_path, set_target!, move_agent!
+
+"""
+    Path{D}
+An alias for `MutableLinkedList{Dims{D}}`. Used to represent the path to be
+taken by an agent in a `D` dimensional [`GridSpace{D}`](@ref).
+"""
+const Path{D} = MutableLinkedList{Dims{D}}
+
+"""
+    CostMetric{D}
+An abstract type representing a metric that measures the approximate cost of travelling
+between two points in a `D` dimensional [`GridSpace{D}`](@ref). A struct with this as its
+base type is used as the `cost_metric` for the [`Pathfinder`](@ref). To define a custom metric,
+define a struct with this as its base type and a corresponding method for [`delta_cost`](@ref).
+"""
+abstract type CostMetric{D} end
+
+struct MooreMetric{D} <: CostMetric{D}
+    direction_costs::Array{Int,1}
```

Comment:
`Vector{Int}`. Syntactic sugar, but we use it throughout the rest of the codebase.